### DISCUSSION TITLE
oci8 + PHP8.2 + Oracle XE

### DISCUSSION THREAD

**Discussion by hormiai76** [ADOdb/ADOdb#1026] [Q&A] [answered]
Hi,
we would like to migrate our application from PHP7.4 to PHP8.2.
We are using oci8 driver to connect to our Oracle XE 11g database and we would like to upgrade it but we are not sure about the compatibilities with the oci8 driver of the adodb 5.22.x versions.
Do you have any plan to be compatible to PHP8.3 too? I would like to know it to update my project documentation.
Regards,
Alberto

**Comment by dregad** [comment, marked as answer]
I am no longer actively using ADOdb on Oracle, so I can't give you first-hand experience here.

However, I do expect the oci8 driver to be fully compatible with PHP 8.2. Over the past couple of years, we have [fixed 10 PHP 8.x related issues](https://github.com/ADOdb/ADOdb/issues?q=label%3APHP8%2CPHP8.1%2CPHP8.2%2CPHP8.3+label%3A%22OCI8+(Oracle)%22), and there are currently no known outstanding problems. I am not aware of any particular issues with PHP 8.3 either.

With regards to Oracle, I'm pretty sure it would work just fine with later versions. I personally used it last with 19c but that was a few years ago. Note that ADOdb just relies on PHP's oci8 driver, please refer to https://www.php.net/manual/en/oci8.requirements.php for further information. 

Should you experience any problems, please open a bug report and well do our best to fix it. Or even better, submit a pull request with a patch if you can.

**Reply by hormiai76** [reply]
Hi @dregad , 
thank you for your answer. 
Do you know someone in the ADOdb project who can help me? It's time for us to upgrade our DB too. For security reasons we need to update the OS and the database version.

Do you know if we can connect to Oracle 21c XE using the latest adodb oci8 driver? Should we install Oracle 18c XE instead of 21c one? Do you know it adodb oci8 driver is compatible with Oracle 23ai (because Oracle23c XE doesn't exists yet)? It seems that Oracle21cXE EOL is 2025...

Regards,
Alberto Nardella

**Reply by dregad** [reply]
As mentioned before, I don't use Oracle anymore so I don't know for sure. It should work, just try it. I would personally go for the latest available version.

**Comment by mnewnham** [comment]
This week I tested ADOdb using PHP 8.3 and Oracle 23ai free. Whilst I have not run an exhaustive test, I did not come across any issues that needed immediate attention. Hope this helps
